scilpy/tractanalysis/features.py

```diff
@@ -35,15 +35,14 @@ def remove_loops_and_sharp_turns(streamlines,
     -------
     A tuple containing
         list of ndarray: the clean streamlines
-        list of ndarray: the list of removed streamlines, if any
+        list of ndarray: the ids of clean streamlines
     """
 
-    loops = []
     streamlines_clean = []
-    for s in streamlines:
-        if tm.winding(s) >= max_angle:
-            loops.append(s)
-        else:
+    ids = []
+    for i, s in enumerate(streamlines):
+        if tm.winding(s) < max_angle:
+            ids.append(i)
             streamlines_clean.append(s)
 
```

Comment:
Some refactoring need here.

If you use this function inside scil_detect_streamline_loops, it's fine because then you use filter_tractogram_data, which will recuperate data_per_point and data_per_streamline. But if you use this function in scil_compute_connectivity, the function _save_if_needed has not recuperated them yet.

I would change this function to take a SFT as input instead of a streamline, and I would deal with data_per_point and data_per_streamline here.

Then, maybe filter_tractogram_data is not needed anymore.

Reply:
I'm not sure sure about that. I made the `filter_tractogram_data` so that it could be used elsewhere, as it often happens in the library that streamlines are selected but the data associated isn't, when it should. Also (but I don't know how relevant this is), I use this function (`remove_loops_and_sharp_turns`) outside of the scilpy and I need the streamline_ids, although I don't want to impose my usage

Reply:
@EmmaRenauld I just realized that this script will completely change in #84. I'll comment on the other PR so that data is kept by _save_if_needed. No need to update it in this PR imo

Reply:
ok